File: back/src/main/java/com/tdevilleduc/urthehero/back/service/IProgressionService.java
package com.tdevilleduc.urthehero.back.service;

import com.tdevilleduc.urthehero.back.model.Progression;

import java.util.List;
import java.util.Optional;

public interface IProgressionService {

    Progression doProgressionAction(Integer personId, Integer storyId, Integer newPageId);
    List<Progression> findByPersonId(Integer personId);
    Optional<Progression> findByPersonIdAndStoryId(Integer personId, Integer storyId);
    Long countByStoryId(Integer storyId);

}

File: back/src/main/java/com/tdevilleduc/urthehero/back/service/IProgressionService.kt
package com.tdevilleduc.urthehero.back.service

import com.tdevilleduc.urthehero.back.model.Progression
import java.util.*

interface IProgressionService {
    fun doProgressionAction(userId: Int, storyId: Int, newPageId: Int): Progression
    fun findByUserId(userId: Int): MutableList<Progression>
    fun findByUserIdAndStoryId(userId: Int, storyId: Int): Optional<Progression>
    fun countUsersByStoryId(storyId: Int): Long
}